docs: expand baseJobTemplate documentation

kingsley-123 opened the pull request:
### Summary

Fixes #444.

Added comprehensive examples for environment variables, secret references, and image pull secrets configuration in the baseJobTemplate.

### Requirements

- [ ] [Contributing guide](https://github.com/PrefectHQ/prefect-helm/?tab=readme-ov-file#contributing) has been read
- [x] Title follows the [conventional commits](https://www.conventionalcommits.org) format
- [x] Body includes `Closes <issue>`, if available
- [ ] Added/modified configuration includes a descriptive comment for documentation generation
- [ ] Unit tests are added/updated
- [ ] Deprecation/removal checks are added in `templates/NOTES.txt`
- [x] Relevant labels are added
- [x] `Draft` status is used until ready for review

kingsley-123 committed:
Docs: Expand baseJobTemplate documentation with examples for environment variables and secrets #444

kingsley-123 commented:
@mitchnielsen 

Hi,

Thanks for reviewing my PR and for your thoughtful feedback. I appreciate the suggestion about potentially moving this documentation to the main Prefect repository for wider visibility.

I agree that these examples would benefit all users, not just those using the Helm charts. I'm happy to take either approach:

1. Continue with this PR to improve the Helm-specific documentation
2. Close this PR and create a new one in the main Prefect repository with expanded examples
3. Do both - keep simpler notes here while creating more comprehensive documentation in the main repo

I also noticed one of the tests is failing for version 1.31.0 while passing for newer versions. Would you happen to have any logs or insights on what might be causing this issue?

Please let me know which direction you prefer, and I'll be glad to make the necessary adjustments. I'm committed to improving the documentation wherever it will be most helpful to users.

Thanks again for your guidance.

kingsley-123 committed:
Clarify base job template usage in README

Simplify baseJobTemplate documentation as requested

mitchnielsen commented on charts/prefect-worker/README.md.gotmpl:
```diff
@@ -245,35 +245,36 @@ helm install prefect-worker prefect/prefect-worker -f values.yaml --set-file wor
 the snippet is truncated for brevity. The full configuration should still be
 provided.
 
+For comprehensive examples including environment variables, secret references, resource configurations, and advanced use cases, see the [Prefect base job template documentation](https://docs.prefect.io/latest/guides/deployment/kubernetes/#base-job-templates).
```
I don't think this link is working for me. I wonder if we want one or both of these:

- https://docs.prefect.io/v3/how-to-guides/deployment_infra/kubernetes#configure-work-pool-options
- https://docs.prefect.io/v3/how-to-guides/deployment_infra/manage-work-pools#base-job-template

kingsley-123 committed:
fix: correct documentation link and code block formatting

kingsley-123 commented:
@mitchnielsen 

Hi, I've updated the PR with your suggested changes:
- Fixed the documentation link to point to the correct URL
- Corrected the code block formatting

Ready for another review. Thanks.
